debezium-server-voyager/debezium-server-voyagerexporter/src/main/java/io/debezium/server/ybexporter/YbExporterConsumer.java

```diff
@@ -80,6 +93,97 @@ void connect() throws URISyntaxException {
         flusherThread = new Thread(this::flush);
         flusherThread.setDaemon(true);
         flusherThread.start();
+
+        // Register shutdown hook to release lock on exit
```

Comment:
will the shutdown hook also run if there is some exception thrown? 
If so, let's move this block above immediately after line 71. That way,  even if there is some exception in the lines after that, it will gracefully release the lock.

Reply:
yes